### PR TITLE
Future iterators.

### PR DESCRIPTION
**Pull request by dvc94ch**

### CODE REVIEW

File: src/abi.rs [hunk 1]
```diff
@@ -305,11 +305,19 @@ impl Interface {
...
                     match p {
                         AbiType::Option(ty) | AbiType::Result(ty) => p = &**ty,
+                        AbiType::Future(ty) => {
+                            symbol.push_str("_future_poll");
+                            p = &**ty
+                        }
+                        AbiType::Stream(ty) => {
+                            symbol.push_str("_stream_poll");
+                            p = &**ty
+                        }
```
Comment (wngr) [inline src/abi.rs:319]: Why's that necessary?

**Review by wngr — COMMENTED** [review]
So whenever an `Iterator` is nested within a `Future` or `Stream`, it needs to be a Vec?

File: src/abi.rs (hunk 1 above)
Comment (dvc94ch) [reply src/abi.rs:319]: it's the next function of the iterator returned by the poll method.

**Comment by dvc94ch** [comment]
Actually it always needs to be a vec now

**Review by wngr — APPROVED** [review]

**Pull request merged**